liquibase-core/src/main/java/liquibase/Liquibase.java

```diff
@@ -260,6 +260,7 @@ public void update(Contexts contexts, LabelExpression labelExpression, boolean c
      * <p>
      * NOTE: to reduce the number of queries to the databasehistory table, this method will cache the "fast check" results within this instance under the assumption that the total changesets will not change within this instance.
      */
+    @Deprecated
     protected boolean isUpToDateFastCheck(Contexts contexts, LabelExpression labelExpression) throws LiquibaseException {
         String cacheKey = contexts +"/"+ labelExpression;
         if (!this.upToDateFastCheck.containsKey(cacheKey)) {
```

Comment:
Can you remove this code and replace it with a call to UpdateCommandStep.isUpToDateFastCheck() ?

Reply:
Sure, leave the method. But this method could call UpdateCommandStep.isUpToDateFastCheck() instead of duplicating the logic.

Reply:
Yes, please :)

Reply:
Done!